packages/flame/lib/src/test_helpers/mock_canvas.dart

```diff
@@ -115,6 +118,34 @@ class MockCanvas extends Fake implements Canvas, Matcher {
     return mismatchDescription.add(matchState['description'] as String);
   }
 
+  bool containsAtLeast(covariant MockCanvas other, Map matchState) {
+    if (_saveCount != 0) {
+      return _fail('Canvas finished with saveCount=$_saveCount', matchState);
+    }
+    final useTolerance = max(tolerance, other.tolerance);
+    final remainingExpectedCommands = other._commands.toList();
+    for (final expectedCommand in _commands) {
+      final idx = remainingExpectedCommands.indexWhere((cmd) {
+        if (expectedCommand.runtimeType != cmd.runtimeType) {
+          return false;
+        }
+
+        expectedCommand.tolerance = cmd.tolerance = useTolerance;
+        return expectedCommand.equals(cmd);
+      });
```

Comment:
that might not work too well if you are expecting multiple equal commands though right? bc since they can be in any order I don't want to end up matching them by type in a way that then leads to an error where if I happen to choose a better matching by content it would actually work out